crypto/fipsmodule/digest/digest.c

```diff
@@ -269,8 +281,16 @@ int EVP_Digest(const void *data, size_t count, uint8_t *out_md,
 
   EVP_MD_CTX_init(&ctx);
   ret = EVP_DigestInit_ex(&ctx, type, impl) &&
-        EVP_DigestUpdate(&ctx, data, count) &&
-        EVP_DigestFinal_ex(&ctx, out_md, out_size);
+        EVP_DigestUpdate(&ctx, data, count);
+  if (EVP_MD_flags(type) & EVP_MD_FLAG_XOF) {
```

Comment:
can we extract `EVP_MD_flags(type) & EVP_MD_FLAG_XOF` to a function `evp_md_is_xof` and call the function instead of having the expression?

Reply:
Hm, my view is that this pattern is common, safe, and simple enough that it doesn't really justify its own function. I don't feel very strongly about this, though. What do you think?